API/API.mli

```diff
@@ -2294,73 +2389,83 @@ sig
     | UnivRigid
     | UnivFlexible of bool
 
-    type 'a freelisted = {
-          rebus : 'a;
-          freemetas : Metaset.t
-        }
+  val univ_rigid : rigid
+
+  type 'a freelisted = {
+    rebus : 'a;
+    freemetas : Metaset.t
+  }
+
+  type instance_constraint = IsSuperType | IsSubType | Conv
+
+  type instance_typing_status =
+      CoerceToType | TypeNotProcessed | TypeProcessed
+
+  type instance_status = instance_constraint * instance_typing_status
+
+  type clbinding =
+    | Cltyp of Names.Name.t * Constr.t freelisted
+    | Clval of Names.Name.t * (Constr.t freelisted * instance_status) * Constr.t freelisted
 
```

Comment:
My main concern with comments is that they're meaningful only to humans. And I'm not sure people are actually going to read the API mli file because it lacks documentation. I'd favour some warning-based solution, but then that's going to spit a lot of warnings in plugins and people are going to complain.